cdc/sinkv2/ddlsink/mysql/mysql_ddl_sink.go

```diff
@@ -97,6 +97,10 @@ func NewMySQLDDLSink(
 
 func (m *mysqlDDLSink) WriteDDLEvent(ctx context.Context, ddl *model.DDLEvent) error {
 	err := m.execDDLWithMaxRetries(ctx, ddl)
+	// we should not retry changefeed if DDL failed by return an unretryable error.
+	if !errorutil.IsRetryableDDLError(err) {
```

Comment:
My idea is that when encounter a non-retryable error , we can wrap it as ChangefeedUnretryableErr ASAP. When it is thrown to the owner, it can be handled correctly.